4/n Gemini Prompt Schema

Ankush-lastmile opened the pull request:
4/n Gemini Prompt Schema

Prompt Schema for Gemini Pro.

This prompt schema doesn't define a schema for Gemini's Safety Settings As those are more complicated, they will come on top.

API Docs for Generation Config: https://ai.google.dev/api/python/google/ai/generativelanguage/GenerationConfig

Skipped these attributes as part of the prompt schema.
- skipped candidate count
- didn't add max_output_tokens. I was getting weird index out of bounds exceptions when setting max_output_tokens which occured at the output parsing step. I tried with 35, 400, 200. Thought it would be better UX to just leave it out for now before figuring out why.
- safety settings

See:
Code Ref: https://github.com/google/generative-ai-python/blob/main/google/generativeai/types/safety_types.py#L218-L221
API Doc Ref: https://ai.google.dev/docs/safety_setting_gemini#code-examples

## Testplan

<img width="1310" alt="Screenshot 2024-02-26 at 11 20 04 AM" src="https://github.com/lastmile-ai/aiconfig/assets/141073967/9572a4c0-2005-4e6c-9e93-c1eb0c3859dd">

Ankush-lastmile commented:
https://media3.giphy.com/media/v1.Y2lkPTc5MGI3NjExeWt6NTF3N2R1ZzZtY3Q1dmp2ODdnajkxMXU3cDR0cnYyM3g1bzBseiZlcD12MV9naWZzX3NlYXJjaCZjdD1n/PjTkXVXx72KiCMN9W2/200.webp

rossdanlm commented:
Why are we skipping candidate count?

rossdanlm commented on python/src/aiconfig/editor/client/src/shared/prompt_schemas/GeminiPromptSchema.ts:
```diff
@@ -0,0 +1,60 @@
+          },
+        },
+      },
+      safety_settings: {
+        type: "object",
+        description:
+          "Passing a safety setting for a category changes the allowed proability that content is blocked.",
+      },
+    },
```
Let's just not include here if it's never showing up in the first place

Ankush-lastmile replied:
removed

rossdanlm commented on python/src/aiconfig/editor/client/src/utils/promptUtils.ts:
```diff
@@ -23,6 +23,7 @@ import { HuggingFaceTextTranslationRemoteInferencePromptSchema } from "../shared
 import { HuggingFaceImage2TextRemoteInferencePromptSchema } from "../shared/prompt_schemas/HuggingFaceImage2TextRemoteInferencePromptSchema";
 import { ClaudeBedrockPromptSchema } from "../shared/prompt_schemas/ClaudeBedrockPromptSchema";
 import { HuggingFaceConversationalRemoteInferencePromptSchema } from "../shared/prompt_schemas/HuggingFaceConversationalRemoteInferencePromptSchema";
+import { GeminiParserPromptSchema } from "../shared/prompt_schemas/GeminiPromptSchema";
```
Let's do alphabetical or the order in which the models appear

Ankush-lastmile commented:
- changed location of import
- removed safety settings for now. Will add later once defined in schema

>Why are we skipping candidate count?
Sorry should have added to pr description. Setting it to more than 1 throws an error. Gemini might not support multiple candidates, will have to investigate
